From fbda66c2589173706a7ec8d386a84c44cb8f449e Mon Sep 17 00:00:00 2001
From: Janne Grunau <j@jannau.net>
Date: Fri, 31 Oct 2025 09:56:58 +0100
Subject: [PATCH] hk: Report the correct plane count in
 VkDrmFormatModifierProperties2?EXT

Fixes import of planar formats like NV12 in gtk4. Allows
`gst-launch-1.0 v4l2src ! gtk4paintablesink` to use vulkan instead of
falling back to OpenGL.

Closes: #14217
Cc: mesa-stable
Signed-off-by: Janne Grunau <j@jannau.net>
---
 src/asahi/vulkan/hk_format.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/asahi/vulkan/hk_format.c b/src/asahi/vulkan/hk_format.c
index 250eb2a70f9f1..006e3cd05a4d8 100644
--- a/src/asahi/vulkan/hk_format.c
+++ b/src/asahi/vulkan/hk_format.c
@@ -67,7 +67,7 @@ get_drm_format_modifier_properties_list(
       {
          *out_props = (VkDrmFormatModifierPropertiesEXT){
             .drmFormatModifier = mod,
-            .drmFormatModifierPlaneCount = 1 /* no planar mods */,
+            .drmFormatModifierPlaneCount = vk_format_get_plane_count(vk_format),
             .drmFormatModifierTilingFeatures = flags,
          };
       };
@@ -96,7 +96,7 @@ get_drm_format_modifier_properties_list_2(
       {
          *out_props = (VkDrmFormatModifierProperties2EXT){
             .drmFormatModifier = mod,
-            .drmFormatModifierPlaneCount = 1, /* no planar mods */
+            .drmFormatModifierPlaneCount = vk_format_get_plane_count(vk_format),
             .drmFormatModifierTilingFeatures = flags,
          };
       };
-- 
GitLab

